numpy/linalg/linalg.py

```diff
@@ -2149,7 +2149,9 @@ def norm(x, ord=None, axis=None, keepdims=False):
         if not (-nd <= row_axis < nd and -nd <= col_axis < nd):
             raise ValueError('Invalid axis %r for an array with shape %r' %
                              (axis, x.shape))
-        if row_axis % nd == col_axis % nd:
```

Comment:
What is normally done elsewhere would look something like:

```
if row_axis < 0:
    row_axis += nd
if col_axis < 0:
    col_axis += nd
if  not(0 <= row_axis < nd and 0 <= col_axis < nd ):
    raise ValueError('Blah, blah, blah')
if row_axis == col_axis:
    raise ValueError('Blah, blah, blah')
```

Not that your solution doesn't work, but having a consistent approach to similar things makes it easier to understand the code.